shared-module/displayio/Display.c

```diff
@@ -136,7 +136,9 @@ void common_hal_displayio_display_construct(displayio_display_obj_t *self,
 
     // Set the group after initialization otherwise we may send pixels while we delay in
     // initialization.
-    common_hal_displayio_display_set_root_group(self, &circuitpython_splash);
+    if (!circuitpython_splash.in_group) {
```

Comment:
This code looks fine, but I noticed a few lines later it checks if `root_group` is NULL and if so assigns it to `circuitpython_splash` with no checks and not using the function. Not sure how that may affect anything. But your check on its own looks good.

Reply:
Are you talking about when calling `display.show(None)` ? It uses `common_hal_displayio_display_show` which does a few checks, including if the group is already in a group, and returns False when those checks fail. That return is checked and an exception is raised in the shared-bindings part:
https://github.com/adafruit/circuitpython/blob/4e7bbf70a67fe15a135ad859fa04c13bcef4fde3/shared-bindings/displayio/Display.c#L261-L264
Maybe it could use `common_hal_displayio_display_set_root_group` instead. (`show()` is deprecated anyway).